model/flow/service_event_test.go

```diff
@@ -14,6 +15,20 @@ import (
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
+// "For best performance, reuse EncMode and DecMode after creating them." [1]
+// [1] https://github.com/fxamacker/cbor
+var cborEncMode = func() cbor.EncMode {
+	options := cbor.CoreDetEncOptions() // CBOR deterministic options
+	// default: "2021-07-06 21:20:00 +0000 UTC" <- unwanted
+	// option : "2021-07-06 21:20:00.820603 +0000 UTC" <- wanted
+	options.Time = cbor.TimeRFC3339Nano // option needed for wanted time format
+	encMode, err := options.EncMode()
+	if err != nil {
+		panic(err)
```

Comment:
@yhassanzadeh13 thanks for the comment!

I refactored the code so that the block of code only exists once instead of three times. This was a useful exercise because the first two refactors failed due to Golang circular dependencies, which was my first tangle with this particular error :-)